packages/flutter_tools/lib/src/fuchsia/fuchsia_build.dart

```diff
@@ -49,12 +53,79 @@ Future<void> buildFuchsia({
   await _timedBuildStep('fuchsia-kernel-compile',
     () => fuchsiaSdk.fuchsiaKernelCompiler.build(
       fuchsiaProject: fuchsiaProject, target: target, buildInfo: buildInfo));
+
+  if (buildInfo.usesAot) {
+    await _timedBuildStep('fuchsia-gen-snapshot',
+      () => _genSnapshot(fuchsiaProject, target, buildInfo, targetPlatform));
+  }
+
   await _timedBuildStep('fuchsia-build-assets',
     () => _buildAssets(fuchsiaProject, target, buildInfo));
   await _timedBuildStep('fuchsia-build-package',
     () => _buildPackage(fuchsiaProject, target, buildInfo, runnerPackageSource));
 }
 
+Future<void> _genSnapshot(
+  FuchsiaProject fuchsiaProject,
+  String target, // lib/main.dart
+  BuildInfo buildInfo,
+  TargetPlatform targetPlatform,
+) async {
+  final String outDir = getFuchsiaBuildDirectory();
+  final String appName = fuchsiaProject.project.manifest.appName;
+  final String dilPath = fs.path.join(outDir, '$appName.dil');
+
+  final String vmSnapshotData = fs.path.join(outDir, 'vm_data.aotsnapshot');
+  final String vmSnapshotInstructions = fs.path.join(outDir, 'vm_instructions.aotsnapshot');
+  final String snapshotData = fs.path.join(outDir, 'data.aotsnapshot');
+  final String snapshotInstructions = fs.path.join(outDir, 'instructions.aotsnapshot');
+
+  final String genSnapshot = artifacts.getArtifactPath(
+    Artifact.genSnapshot,
+    platform: targetPlatform,
+    mode: buildInfo.mode,
+  );
+
+  final List<String> flags = <String>[
+    '--no_causal_async_stacks',
+    '--deterministic',
+    '--snapshot_kind=app-aot-blobs',
+    '--vm_snapshot_data=$vmSnapshotData',
+    '--vm_snapshot_instructions=$vmSnapshotInstructions',
+    '--isolate_snapshot_data=$snapshotData',
+    '--isolate_snapshot_instructions=$snapshotInstructions',
+    if (buildInfo.isDebug) '--enable-asserts',
+    dilPath,
+  ];
+
+  final List<String> command = <String>[
+    genSnapshot,
+    ...flags,
+  ];
+  final Process process = await processUtils.start(command);
```

Comment:
I believe this logic is the same as `processUtils.stream` with `trace: true`

Reply:
Good call. Done.